internal/bridge/crunchybridgecluster/postgres_test.go

```diff
@@ -0,0 +1,139 @@
+
+	"gotest.tools/v3/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/crunchydata/postgres-operator/internal/bridge"
+	"github.com/crunchydata/postgres-operator/internal/testing/require"
+	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
+)
+
+func TestGeneratePostgresRoleSecret(t *testing.T) {
+	_, tClient := setupKubernetes(t)
+	require.ParallelCapacity(t, 0)
+
+	reconciler := &CrunchyBridgeClusterReconciler{Client: tClient}
+
+	cluster := testCluster()
+	cluster.Namespace = setupNamespace(t, tClient).Name
+
+	spec := &v1beta1.CrunchyBridgeClusterRoleSpec{
+		Name:       "application",
+		SecretName: "application-role-secret",
+	}
+	role := &bridge.ClusterRoleApiResource{
+		Name:     "application",
+		Password: "password",
+		URI:      "postgres://application:password@p.33ddusyv3fgkrf2bz67bxlkpwa.db.postgresbridge.com:5432/postgres",
+	}
+	t.Run("ObjectMeta", func(t *testing.T) {
+		secret, err := reconciler.generatePostgresRoleSecret(cluster, spec, role)
+		assert.NilError(t, err)
+
+		if assert.Check(t, secret != nil) {
+			assert.Equal(t, secret.Namespace, cluster.Namespace)
+			assert.Assert(t, metav1.IsControlledBy(secret, cluster))
+			assert.DeepEqual(t, secret.Labels, map[string]string{
+				"postgres-operator.crunchydata.com/cluster":    "hippo-cr",
+				"postgres-operator.crunchydata.com/role":       "cbc-pgrole",
+				"postgres-operator.crunchydata.com/cbc-pgrole": "application",
+			})
+		}
+	})
+
+	t.Run("Data", func(t *testing.T) {
+		secret, err := reconciler.generatePostgresRoleSecret(cluster, spec, role)
+		assert.NilError(t, err)
+
+		if assert.Check(t, secret != nil) {
+			assert.Equal(t, secret.StringData["name"], "application")
+			assert.Equal(t, secret.StringData["password"], "password")
+			assert.Equal(t, secret.StringData["uri"],
+				"postgres://application:password@p.33ddusyv3fgkrf2bz67bxlkpwa.db.postgresbridge.com:5432/postgres")
+		}
+	})
+}
+
+func TestReconcilePostgresRoleSecrets(t *testing.T) {
+	ctx := context.Background()
+	_, tClient := setupKubernetes(t)
+	require.ParallelCapacity(t, 0)
+
+	reconciler := &CrunchyBridgeClusterReconciler{Client: tClient}
+
+	apiKey := "1234567890"
+	ns := setupNamespace(t, tClient).Name
+
+	t.Run("DuplicateSecretNameInSpec", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns
+
+		spec1 := &v1beta1.CrunchyBridgeClusterRoleSpec{
+			Name:       "application",
+			SecretName: "role-secret",
+		}
+		spec2 := &v1beta1.CrunchyBridgeClusterRoleSpec{
+			Name:       "postgres",
+			SecretName: "role-secret",
+		}
+		cluster.Spec.Roles = append(cluster.Spec.Roles, spec1, spec2)
+
+		roleSpecSlice, secretMap, err := reconciler.reconcilePostgresRoleSecrets(ctx, apiKey, cluster)
+		assert.Check(t, roleSpecSlice == nil)
+		assert.Check(t, secretMap == nil)
+		assert.ErrorContains(t, err, "There are duplicate Role SecretNames in the spec. SecretNames must be unique.",
```

Comment:
So the message is saying that the CrunchyBridgeCluster spec has duplicate Role SecretNames (i.e. two or more Roles have the same SecretName)... I could maybe reword it to: "Two or more of the Roles in the CrunchyBridgeCluster spec have the same SecretName. Role SecretNames must be unique." How does that sound?

Reply:
I think it would be fine to just add the type of spec to the original message.
```suggestion
		assert.ErrorContains(t, err, "There are duplicate Role SecretNames in the CrunchyBridgeCluster spec. SecretNames must be unique.",
```